### ISSUE TITLE
the XHR server is unavailable or missing

### ISSUE THREAD

**Issue by udisun** [cypress-io/cypress#510]
* Operating System: mac osx
* Cypress Version: 0.19.2
* Browser/Browser Version: chrome/58

### Are you requesting a feature or reporting a bug?
reporting a bug

### Current behavior:
using `cy.server()` inside a command being called from `before all` is throwing an error

### Expected behavior:
should work as documented

### How to reproduce the current behavior:
run the code i've written done in the "test code" section

#### Test code:

```js
describe('e2e testing', () => {
  before(() => {
    cy.deleteUser()
  })
  it('signup and click', () => {
    cy
      .server()
      .route(/\//).as('signup')
      .signup()
      .wait('@signup')
      .get('side-bar play-button').click()
  })
})

Cypress.addParentCommand('deleteUser', (username, password) => {
  const log = Cypress.Log.command({
    name: 'deleteUser',
    message: [username, password],
    consoleProps: () => ({
      username,
      password,
    }),
  })

  cy
    .server()
    .route(/\//).as('login')
    .login(username, password) // login as the test user
    .wait('@login')
    .request('/api/users/some-path-to-nowhere') // get the test user entityId
    .then(res => {
      // some business logic testing
    })
})
```

### Additional Info (images, notes, stack traces, etc)
```
 CypressError: The XHR server is unavailable or missing. This should never happen and likely is a bug. Open an issue if you see this message.
      at Object.cypressErr (https://zengaming.com/__cypress/static/js/cypress.js:5723:15)
      at Object.throwErr (https://zengaming.com/__cypress/static/js/cypress.js:5688:22)
      at Object.throwErrByPath (https://zengaming.com/__cypress/static/js/cypress.js:5715:21)
      at $Cy.unavailableErr (https://zengaming.com/__cypress/static/js/cypress.js:12621:28)
      at $Cy.getXhrServer (https://zengaming.com/__cypress/static/js/cypress.js:12691:75)
      at $Cy.server (https://zengaming.com/__cypress/static/js/cypress.js:12850:21)
      at https://zengaming.com/__cypress/static/js/cypress.js:1520:37
      at tryCatcher (https://zengaming.com/__cypress/static/js/vendor.js:46486:31)
      at Promise._settlePromiseFromHandler (https://zengaming.com/__cypress/static/js/vendor.js:44621:31)
      at Promise._settlePromiseAt (https://zengaming.com/__cypress/static/js/vendor.js:44697:18)
      at Promise._settlePromises (https://zengaming.com/__cypress/static/js/vendor.js:44813:14)
      at Async._drainQueue (https://zengaming.com/__cypress/static/js/vendor.js:42180:16)
      at Async._drainQueues (https://zengaming.com/__cypress/static/js/vendor.js:42190:10)
      at MutationObserver.Async.drainQueues (https://zengaming.com/__cypress/static/js/vendor.js:42018:14
```

**Comment by jennifer-shehane** [comment]
The issue as stated does not seem to be the case, as I can write the `before` below in my test code and receive no error. 

```
  before(() => {
    cy.server()
  })
```

Is it possible to reduce your use case further? Where is the error occurring within the commands in the Command Log (when running your tests in a browser)? I see you are using another custom command 'login', could you also include the code within that custom command?

**Comment by udisun** [comment]
@jennifer-shehane you missed something
i specifically wrote calling a command that calls server.
I actually got around the issue with a hack.
the error called on my 5th test, while each two tests were wrapped with a `context` call.
for some reason in that structure the 5th call failed with the error i've attached while the same command was being called also from my 1st command successfully.
at the end what solved the issue for me is separating each `context` into another spec file.
and for some reason running the test this way didn't throw the error again.
so it seems to me that the server get's lost in the way when running 5 tests in the same file but when running 2 tests in each file it exists.
wierd :(

**Comment by jennifer-shehane** [comment]
Yes, I see the issue as you describe it now. I wonder if your 5th test also had a call to `cy.server()` in a `before`?

I have been able to confirm that #80 is a regression that seems to have popped up again in a recent release, perhaps it is related?

**Comment by brian-mann** [comment]
The code for this is done, but this has yet to be released. We'll update the issue and reference the changelog when it's released.

**Comment by brian-mann** [comment]
Fixed in [`0.20.0`](https://docs.cypress.io/guides/references/changelog.html#0-20-0)
